File: VisaD.Application/Applications/Dtos/DiplomaFileDto.cs
using System;
using System.Collections.Generic;
using VisaD.Application.Nomenclatures.Dtos;
using VisaD.Data.Applications.Diplomas;
using VisaD.Data.Nomenclatures;

namespace VisaD.Application.Applications.Dtos
{
	public class DiplomaFileDto
	{
		public int Id { get; set; }
		public string DiplomaNumber { get; set; }

		public DateTime IssuedDate { get; set; }

		public NomenclatureDto<Country> Country { get; set; }
		public string City { get; set; }

		public string OrganizationName { get; set; }

		public DiplomaTypeNomenclatureDto Type { get; set; }

		public DiplomaDocumentFile DiplomaDocumentFile { get; set; }

		public List<DiplomaDocumentFile> AttachedFiles { get; set; } = new List<DiplomaDocumentFile>();
	}
}

File: WebApplication/src/modules/application/models/diploma-file.dto.ts
import { NomenclatureDto } from 'src/modules/nomenclature/common/models/nomenclature-dto';
import { DiplomaTypeNomenclatureDto } from './diploma-type-nomenclature.dto';
import { DiplomaDocumentFile } from './diplomas/diploma-document-file';

export class DiplomaFileDto {
	id: number;
	diplomaNumber: string;
	issuedDate: Date | null;
	country: NomenclatureDto;
	city: string;
	type: DiplomaTypeNomenclatureDto;
	organizationName: string;
	diplomaDocumentFile: DiplomaDocumentFile;
	attachedFiles: DiplomaDocumentFile[] = [];

	disabled: boolean = false;
	showNacidRecommendation: boolean = false;
	canEdit: boolean = true;
}
